grid/save-state-in-wasm-through-controller/GridSavingStateInController/Client/Pages/Index.razor

```diff
@@ -0,0 +1,148 @@
+<br />
+<br />
+
+<TelerikButton OnClick="@SaveState">Save State</TelerikButton>
+<TelerikButton OnClick="@ApplyState">Apply last saved State</TelerikButton>
+
+<TelerikGrid Data="@GridData" Height="500px" @ref="@Grid"
+             Groupable="true"
+             Pageable="true"
+             Sortable="true"
+             FilterMode="@GridFilterMode.FilterRow"
+             Reorderable="true"
+             Resizable="true"
+             SelectionMode="GridSelectionMode.Multiple" @bind-SelectedItems="@SelectedItems"
+             OnUpdate=@UpdateItem OnDelete=@DeleteItem OnCreate=@CreateItem EditMode="@GridEditMode.Inline">
+    <GridColumns>
+        <GridColumn Field="@(nameof(SampleData.Id))" Editable="false" />
+        <GridColumn Field="@(nameof(SampleData.Name))" Title="Employee Name" />
+        <GridColumn Field="@(nameof(SampleData.Team))" Title="Team" />
+        <GridCommandColumn>
+            <GridCommandButton Command="Edit" Icon="edit">Edit</GridCommandButton>
+            <GridCommandButton Command="Delete" Icon="delete">Delete</GridCommandButton>
+            <GridCommandButton Command="Save" Icon="save" ShowInEdit="true">Save</GridCommandButton>
+            <GridCommandButton Command="Cancel" Icon="cancel" ShowInEdit="true">Cancel</GridCommandButton>
+        </GridCommandColumn>
+    </GridColumns>
+    <GridToolBar>
+        <GridCommandButton Command="Add" Icon="add">Add Employee</GridCommandButton>
+    </GridToolBar>
+</TelerikGrid>
+
+@if (SelectedItems != null)
+{
+    <ul>
+        @foreach (SampleData employee in SelectedItems)
+        {
+            <li>
+                @employee.Id
+            </li>
+        }
+    </ul>
+}
+
+@code {
+    TelerikGrid<SampleData> Grid { get; set; }
+    IEnumerable<SampleData> SelectedItems { get; set; } = Enumerable.Empty<SampleData>();
+    List<SampleData> GridData { get; set; }
+
+    async void SaveState()
+    {
+        var state = await Http.PostAsJsonAsync("setstate", Grid.GetState());
+    }
+    
+    async void ApplyState()
+    {
+        var state = await Http.GetFromJsonAsync<GridState<SampleData>>("getstate");
+
+        await Grid.SetState(state);
+    }
+
+    async Task UpdateItem(GridCommandEventArgs args)
+    {
+        SampleData item = (SampleData)args.Item;
+
+        await MyService.Update(item);
+
+        await GetGridData();
+    }
+
+    async Task DeleteItem(GridCommandEventArgs args)
+    {
+        SampleData item = (SampleData)args.Item;
+
+        await MyService.Delete(item);
+
+        await GetGridData();
+    }
+
+    async Task CreateItem(GridCommandEventArgs args)
+    {
+        SampleData item = (SampleData)args.Item;
+
+        await MyService.Create(item);
+
+        await GetGridData();
+    }
+
+    async Task GetGridData()
+    {
+        GridData = await MyService.Read();
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        await GetGridData();
+    }
+
+    public static class MyService
```

Comment:
Maybe we should use some more meaningful and intuitive name? It can suggest what the service is for.

Reply:
Done